### DISCUSSION TITLE
Performance benchmarks

### DISCUSSION THREAD

**Discussion by crocodele** [supermetrics/php-ext-jsonpath#90] [Ideas]
I've run some performance benchmarks using [a dataset from Tate Gallery](https://github.com/tategallery/collection). A JSON file consisting of all the artists and artworks metadata contains ~70k rows and is ~60MB in filesize.

So far it's all in a messy standalone script. Using [PHPBench](https://github.com/phpbench/phpbench) could be an option. I haven't used it before, but it looks quite solid. If anyone has other suggestions, I'm happy to hear them. :slightly_smiling_face:

**Comment by crocodele** [comment]
@mkaminski1988, here's an example of a report from PHPBench:

![Screenshot from 2021-05-22 02-55-02](https://user-images.githubusercontent.com/1522694/119208371-96138380-baaa-11eb-8ce3-b01ca5aa3011.png)

I'll add a few more advanced JSONPath expressions as well.

**Reply by mk6i** [reply]
This is very useful. 👍  It'll help figuring out what performance optimizations to make going forward. I think we should aim to make the extension more competitive with the native PHP implementation for smaller datasets.
